src/lib.rs

```diff
@@ -103,6 +103,11 @@
 //! # Ok(())
 //! # }
 //! ```
+
+#![forbid(unsafe_code, future_incompatible, rust_2018_idioms)]
+#![deny(missing_debug_implementations, nonstandard_style)]
+#![warn(missing_docs, unreachable_pub)]
+
```

Comment:
This is the set of lints introduced. This includes linting for `rust_2018_idioms` which are things that are disallowed in the current edition and can prevent us from using rust 2018 in the future. And `future_incompatible` which are things that will be disallowed in a future edition (due next year).